src/Features/CSharp/Portable/QuickInfo/CSharpSyntacticQuickInfoProvider.cs

```diff
@@ -28,18 +29,30 @@ public CSharpSyntacticQuickInfoProvider()
         protected override Task<QuickInfoItem?> BuildQuickInfoAsync(
             QuickInfoContext context,
             SyntaxToken token)
-            => Task.FromResult(BuildQuickInfo(token));
+            => Task.FromResult(BuildQuickInfo(token, context.CancellationToken));
 
         protected override Task<QuickInfoItem?> BuildQuickInfoAsync(
             CommonQuickInfoContext context,
             SyntaxToken token)
-            => Task.FromResult(BuildQuickInfo(token));
+            => Task.FromResult(BuildQuickInfo(token, context.CancellationToken));
 
-        private static QuickInfoItem? BuildQuickInfo(SyntaxToken token)
+        private static QuickInfoItem? BuildQuickInfo(SyntaxToken token, CancellationToken cancellationToken)
         {
-            if (token.Kind() != SyntaxKind.CloseBraceToken)
+            switch (token.Kind())
             {
-                return null;
+                case SyntaxKind.CloseBraceToken:
+                    break;
```

Comment:
The  code following this switch statement is handling the Brace cases. I would refactor this into that into `BuildQuickInfoCloseBrace`. It might be even useful to place the functions `BuildQuickInfoEndRegion` and `BuildQuickInfoCloseBrace` into their own partial classes. What do you think?